src/Polly.Shared/Monkey/AsyncInjectBehaviourSyntax.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Polly.Monkey
+{
+    /// <summary>
+    /// Fluent API for defining Monkey <see cref="Policy"/>. 
+    /// </summary>
+    public partial class MonkeyPolicy
+    {
+        #region Action Delegate Based Monkey Policies
+
+        /// <summary>
+        /// Builds a <see cref="AsyncInjectBehaviourPolicy"/> which executes a behaviour if <paramref name="enabled"/> returns true and
+        /// a random number is within range of <paramref name="injectionRate"/>.
+        /// </summary>
+        /// <param name="behaviour">Behaviour Delegate to be executed without context</param>
+        /// <param name="injectionRate">The injection rate between [0, 1]</param>
+        /// <param name="enabled">Lambda to check if this policy is enabled in context free mode</param>
+        /// <returns>The policy instance.</returns>
+        public static AsyncInjectBehaviourPolicy InjectBehaviourAsync(
+            Func<Task> behaviour,
+            Double injectionRate,
```

Comment:
should we have a check for < 0 value? across the code base

Reply:
>should we have a check for < 0 value? across the code base

Yes, this would be good.  I saw the opportunity to [add some quick boundary checks in the execution-phase](https://github.com/App-vNext/Polly/blob/f7cbbd8313466426cc02e9ba78b8d71c1dd58a3f/src/Polly.Shared/Monkey/AsyncMonkeyEngine.cs#L18-L25) in my contribution, these target the cases where the dynamic `Func<..., double>` variants generate some out-of-range `double` as well.  But for cases where the injection rate is configured as a non-dynamic `Double injectionRate`, configuration-time checking would be an improvement. 

Tagging this mentally to be done after we've moved the codebase over to the new repo.

Reply:
Don't know if I misunderstood, but I was taking a look at that, and I think the validation you put on `ShouldInject` method, cover all scenarios since the syntax classes are ensuring that at the end of the day the `injectionRate` will be always a lambda.